src/ansys/units/quantity.py

```diff
@@ -327,7 +408,7 @@ def _relative_unit_check(
             return Quantity(value=new_value, units=other_units)
         return Quantity(value=new_value, units=new_units)
 
-    def __float__(self):
+    def __float__(self: Quantity[float]) -> float:
```

Comment:
This method kind of confused me not sure if returning self.value is correct but it needs to return a float on all paths

Reply:
The code path will TypeError out if you return None so I thought this would be the best way to do this though interested to hear your opinion on this

Reply:
Please see my latest comment. Thank you.

Reply:
So should it unconditionally call get_si_value? - No